File: practice/virus.py
import math

n, d = map(int, input().split(' '))
x_list = []
y_list = []
v_list = []
v_list.append(0)

for i in range(n):
    x, y = map(int, input().split(' '))
    x_list.append(x)
    y_list.append(y)

def dis(x_list, y_list, m, n):
    return math.sqrt(pow(x_list[m] - x_list[n], 2) + pow(y_list[m] - y_list[n], 2))

def new(v_list, k):
    for j in range(n):
        if j not in v_list:
            if dis(x_list, y_list, k, j) <= d:
                v_list.append(j)
                new(v_list, j)

new(v_list, 0)

v_list = sorted(v_list)

for i in range(n):
    if i in v_list:
        print('Yes')
    else:
        print('No')

File: practice/virus.cpp
#include <iostream>
#include <vector>
#include <cmath>

double distance(int x1, int y1, int x2, int y2) {
    return std::sqrt(std::pow(x1 - x2, 2) + std::pow(y1 - y2, 2));
}

void infect(const std::vector<int>& x_list, const std::vector<int>& y_list, std::vector<bool>& infected, int person, int d) {
    infected[person] = true;
    for (int other_person = 0; other_person < infected.size(); ++other_person) {
        if (!infected[other_person] && distance(x_list[person], y_list[person], x_list[other_person], y_list[other_person]) <= d) {
            infect(x_list, y_list, infected, other_person, d);
        }
    }
}

int main() {
    int n, d;
    std::cin >> n >> d;
    
    std::vector<int> x_list(n);
    std::vector<int> y_list(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> x_list[i] >> y_list[i];
    }
    
    std::vector<bool> infected(n, false);
    infect(x_list, y_list, infected, 0, d);
    
    for (int i = 0; i < n; ++i) {
        if (infected[i]) {
            std::cout << "Yes" << std::endl;
        } else {
            std::cout << "No" << std::endl;
        }
    }
    
    return 0;
}